### PR TITLE
[DataGrid] Improve filters

### PR DESCRIPTION
**Pull request by dtassone**
Fix some of the ticks of #633

### CODE REVIEW

File: packages/grid/data-grid/src/DataGrid.tsx [hunk 1]
```diff
@@ -26,6 +28,7 @@ export type DataGridProps = Omit<
   disableColumnResize?: true;
   disableColumnReorder?: true;
   disableMultipleColumnsSorting?: true;
+  disableMultipleColumnsFiltering?: true;
```
Comment (oliviertassinari) [inline packages/grid/data-grid/src/DataGrid.tsx:31]: Sort ASC

```suggestion
  disableMultipleColumnsFiltering?: true;
  disableMultipleColumnsSorting?: true;
```

File: packages/grid/_modules_/grid/components/tools/Preferences.tsx [hunk 2]
```diff
@@ -52,11 +38,9 @@ const useStyles = makeStyles((theme: Theme) => ({
       flexDirection: 'column',
       overflow: 'auto',
       flex: '1 1',
-      paddingTop: 12,
-      paddingLeft: 12,
     },
-    '& .panelFooter': {
-      padding: 12,
+    '& .panelFooter, .panelHeader': {
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/components/tools/Preferences.tsx:42]: Do you plan to address the issue with the CSS class name later on?

**Review by oliviertassinari — COMMENTED** [review]
I'm not sure if I was supposed to start reviewing it, so I have left a few quick feedback :)

**Review by oliviertassinari — APPROVED** [review]
We are moving in the right direction :)

- Regarding item 6. The close transition is now missing, but it's better than having a jump.
- I have also noticed that the scroll is broken when opening the column menu. This is because of the auto focus on the input.

**Comment by oliviertassinari** [comment]
A point we have started to discuss with @DanailH. I think that we should replace `if (process.env.NODE_ENV === 'production')` for the feature toggle, to a `if (process.env.ENABLE_EXPERIMENTAL)` flag. The issue with NODE_ENV is that it's 'production' in the Netlify preview. We can't easily review PRs, it forces us to check out each time. I'm working on it.

**Comment by dtassone** [comment]
> We are moving in the right direction :)
> 
> * Regarding item 6. The close transition is now missing.
> * I have also noticed that the scroll is broken when opening the column menu

I'm not sure what you mean by the scroll.

can you try again the 6 and let me know if it's ok now :)

**Comment by oliviertassinari** [comment]
> I'm not sure what you mean by the scroll.

Open the documentation, you will see, one negative point for storybook's isolation :p. I'm working https://github.com/mui-org/material-ui-x/pull/635#issuecomment-733158305 so we can share a link. My env is broken so I can't share video yet.

**Comment by oliviertassinari** [comment]
jump

**Comment by dtassone** [comment]
this is caused by the `autoFocus` prop on the TextField, maybe a bug in core?

I refactored and fixed the issue

**Comment by oliviertassinari** [comment]
> this is caused by the autoFocus prop on the TextField, maybe a bug in core?

autoFocus is a native attribute, actually discouraged for a11y, but anyway. Yes, it could be an issue with the Popper component, but it depends on if we provide the anchor correctly.

**Review by DanailH — APPROVED** [review]

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**